### PR TITLE
Use ion.Timestamp instead of time.Time and enforce float64 precision

### PR DESCRIPTION
**Pull request by justing-bq**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: base_serializer.go [hunk 1]
```diff
@@ -211,22 +212,37 @@ func (baseSerializer *baseSerializer) getBytes(ionType ion.Type, ionValue interf
...
-		return buf.Bytes()[4:], nil
+	err = writer.Finish()
+	if err != nil {
+		return nil, err
 	}
+
+	bytes := buf.Bytes()[4:]
+
+	if ionType == ion.FloatType && len(bytes) == 5 {
+		// We got back a float32 but we want to hash it as a float64.
```
Comment (matthewh-BQ) [inline base_serializer.go:233]: I'd suggest making a reference to the Ion spec here: https://amzn.github.io/ion-hash/docs/spec.html#4-float

**Comment by therapon** [comment]
Should we uncomment some of the tests that relied on timestamp as well?

File: base_serializer.go (hunk 1 above)
Comment (justing-bq) [reply base_serializer.go]: Done.

**Comment by matthewh-BQ** [comment]
Title seems ambiguous still. Consider "Use ion.Timestamp instead of time.Time and enforce float64 precision".

**Pull request merged**